core/rawdb/rawtemporaldb/accessors_receipt.go

```diff
@@ -37,6 +37,24 @@ func ReceiptAsOfWithApply(tx kv.TemporalTx, txNum uint64, rawLogs types.Logs, tx
 	return r, nil
 }
 
+func FillLogsWithCalculatedFields(tx kv.TemporalTx, txNum uint64, rawLogs types.Logs, txnIdx int, blockHash common.Hash, blockNum uint64, txn types.Transaction) (types.Logs, error) {
```

Comment:
also could be a method of `types.Logs` but instead of tx it will receive `firstLogIndexWithinBlock` along with other args. So db fetching will be one thing (ReceiptAsOf) and field filling is another

Reply:
like  `types.Logs.FillWith(...)`